### ISSUE TITLE
Hunter's Yaml-CPP Version Can't Compile In VS2017

### ISSUE THREAD

**Issue by MatthewJM96** [RegrowthStudios/Vorb#116]
We need to keep an eye on [yaml-cpp](https://github.com/jbeder/yaml-cpp/) for its next release. When it releases we then need to get Hunter to accept that new version as an option for building with. This will enable compilation on VS2017.

Alternatively, we could heed the issues raised by Cristian in [this issue](https://github.com/RegrowthStudios/Vorb/issues/69) and look for an alternative yaml solution.

**Comment by czaloj** [comment]
Yeah, I don't remember what the exact problem was with it :/

**Comment by caseymcc** [comment]
or better yet who needs yaml, I will gladly replace it with json

**Comment by caseymcc** [comment]
In fact, I have already been working on python scripts to convert the current texture yaml files to json, for my needs.

**Comment by Barnold1953** [comment]
+1 for json

**Comment by MatthewJM96** [comment]
Did we not prefer YAML for the ease of understanding for modders and designers? If it makes life significantly easier then JSON gets a +1 from me on that reason, but as it stands yaml-cpp is only preventing compilation on VS2017 due to a single macro - could we not make a request to the yaml-cpp guys to back-propagate their hotfix for that to a new minor release for 0.5?

**Comment by MatthewJM96** [comment]
I definitely prefer the yaml format on grounds of the end-user.

**Comment by Barnold1953** [comment]
yeah we picked YAML for ease of use, and robustness. It is probably better for our needs so if we have a reliable parsing alternative or a way to fix this then its ideal.

**Comment by MatthewJM96** [comment]
I'll make a PR with the yaml-cpp guys/get in touch with them, however they prefer, we can come back to thinking about changing after we know how easy they make it for us to get what we need.

**Comment by bryan-lunt** [comment]
Also cant compile on MacOS.
